src/catalog.rs

```diff
@@ -223,310 +393,271 @@ pub trait TableCatalog: Sync + Send {
                 RepositoryError::UniqueConstraintViolation(_) => {
-                    Error::CollectionAlreadyExists {
-                        name: collection_name.to_string(),
+                    Error::TableAlreadyExists {
+                        name: table_name.to_string(),
                     }
                 }
-                _ => Self::to_sqlx_error(e),
+                RepositoryError::FKConstraintViolation(_) => Error::SchemaDoesNotExist {
+                    name: schema_name.to_string(),
+                },
+                RepositoryError::SqlxError(e) => Error::SqlxError(e),
             })
     }
 
-    async fn create_new_table_version(
+    async fn get(
+        &self,
+        catalog_name: &str,
+        schema_name: &str,
+        table_name: &str,
+    ) -> Result<TableRecord> {
+        match self
+            .repository
+            .get_table(catalog_name, schema_name, table_name)
+            .await
+        {
+            Ok(table) => Ok(table),
+            Err(RepositoryError::SqlxError(sqlx::error::Error::RowNotFound)) => {
+                Err(Error::TableDoesNotExist {
+                    name: table_name.to_string(),
+                })
+            }
+            Err(e) => Err(e.into()),
+        }
+    }
+
+    async fn create_new_version(
         &self,
         uuid: Uuid,
         version: i64,
     ) -> Result<TableVersionId> {
         self.repository
-            .create_new_table_version(uuid, version)
+            .create_new_version(uuid, version)
             .await
             .map_err(|e| match e {
                 RepositoryError::SqlxError(sqlx::error::Error::RowNotFound) => {
                     Error::TableUuidDoesNotExist { uuid }
                 }
-                _ => Self::to_sqlx_error(e),
+                e => e.into(),
             })
     }
 
-    async fn get_all_table_versions(
+    async fn delete_old_versions(
+        &self,
+        catalog_name: &str,
+        schema_name: &str,
+        table_name: &str,
+    ) -> Result<u64, Error> {
+        let table = TableStore::get(self, catalog_name, schema_name, table_name).await?;
+
+        Ok(self.repository.delete_old_versions(table.id).await?)
+    }
+
+    async fn get_all_versions(
         &self,
-        database_name: &str,
+        catalog_name: &str,
         table_names: Option<Vec<String>>,
     ) -> Result<Vec<TableVersionsResult>> {
-        self.repository
-            .get_all_table_versions(database_name, table_names)
-            .await
-            .map_err(Self::to_sqlx_error)
+        Ok(self
+            .repository
+            .get_all_versions(catalog_name, table_names)
+            .await?)
     }
 
-    async fn move_table(
+    async fn update(
         &self,
-        table_id: TableId,
+        old_catalog_name: &str,
+        old_schema_name: &str,
+        old_table_name: &str,
+        _new_catalog_name: &str, // For now we don't support moving across catalogs
```

Comment:
change this to an assert that fails if the names aren't the same in case someone downstream tries to use this?